File: Go/find_index_of_the1st_occurrence_in_a_string/28.go
package findindexofthe1stoccurrenceinastring

func strStr(haystack string, needle string) int {
	needleLength := len(needle)
	haystackLength := len(haystack)
	if needleLength > haystackLength {
		return -1
	}
	for i := 0; i <= haystackLength-needleLength; i++ {
		if haystack[i:i+needleLength] == needle {
			return i
		}
	}
	return -1
}

File: TypeScript/find_index_of_first_occur/28.ts
function strStr(haystack: string, needle: string): number {
  const haystackLen = haystack.length;
  const needleLen = needle.length;
  if (needleLen > haystackLen) return -1;
  for (let i = 0; i <= haystackLen - needleLen; i++) {
    const s = haystack.slice(i, i + needleLen);
    if (s === needle) {
      return i;
    }
  }
  return -1;
}

export default strStr;
